content/understand-django/2020-01-22-urls-lead-way.es.md

```diff
@@ -0,0 +1,444 @@
+)
+from django.urls import reverse
+
+def old_blog_categories(request):
+    return HttpResponseRedirect(
+        reverse("blog_categories")
+    )
+```
+
+El trabajo de `reverse` es buscar cualquier nombre de ruta y devolver su ruta equivalente. Eso significa que:
+
+```python
+reverse("blog_categories") == "/marketing/blog/categories/"
+```
+
+Al menos hasta que quieras cambiarlo de nuevo. 😁
+
+## Cuando los nombres chocan
+
+¿Qué sucede si tienes varias URL a las que deseas dar el mismo nombre (`name`)? Por ejemplo, índice (`index`) o detalle (`detail`) son nombres comunes que quizás desee aplicar. Podemos recurrir a
+{{< extlink "https://www.python.org/dev/peps/pep-0020/" "The Zen of Python" >}}
+para obtener asesoramiento.
+
+> El Zen de Python, de Tim Peters
+>
+> Hermoso es mejor que feo.
+>
+> ...
+>
+> **Los espacios de nombres son una gran idea -- ¡hagamos más de eso!**
+
+Los espacios de nombres pueden ser nuevos para tí si no has estado programando por mucho tiempo. Son un *espacio compartido para los nombres*. Tal vez eso esté claro, pero recuerdo haber luchado con el concepto cuando comencé a escribir software.
+
+Para hacer una analogía con algo en el mundo real, usemos contenedores. Imagina que tienes dos bolas rojas y dos bolas azules. Ponga una bola de cada color en cada uno de los dos contenedores etiquetados "A" y "B". Si quisiera una bola azul específica, no puedo decir "por favor, dame la bola azul" porque sería ambiguo. En cambio, para obtener una bola específica, tendría que decir "por favor, dame la bola azul en el contenedor B". En este escenario, el depósito es el espacio de nombres.
+
+El ejemplo que usamos para escuelas y estudiantes puede ayudar a ilustrar esta idea en código. Ambas aplicaciones tenían una vista de `index` para representar la raíz de las partes respectivas del proyecto (es decir, `schools/` y `students/`). Si quisiéramos referirnos a esas vistas, intentaríamos elegir la opción de índice más fácil. Desafortunadamente, si elige `index`, Django no puede decir cuál es la vista correcta para `index`. El nombre es ambiguo.
+
+Una solución es crear su propio espacio de nombres anteponiendo el nombre (`name`) con algo común como `schools_`. El problema con ese enfoque es que la URLconf se repite.
+
+```python
+# schools/urls.py
+from django.urls import path
+
+from schools import views
+
+urlpatterns = [
+    path(
+        "",
+        views.index,
+        name="schools_index"
+    ),
+    path(
+        "<int:school_id>/",
+        views.school_detail,
+        name="schools_detail"
+    ),
+]
+```
+
+Django proporciona una alternativa que te permitirá mantener un nombre más corto.
+
+```python
+# schools/urls.py
+from django.urls import path
+
+from schools import views
+
+app_name = "schools"
+urlpatterns = [
+    path("", views.index, name="index"),
+    path(
+        "<int:school_id>/",
+        views.school_detail,
+        name="detail"
+    ),
+]
+```
+
+By adding `app_name`,
```

Comment:
This section was missing the translation in the Google Doc.

Reply:
Al añadir app_name,

Reply:
Sorry, I know you haven't done much with PRs so I should have been more clear. The "section" is lines 412-419. It appears that the translation you listed is just for 412.

Reply:
Al añadir app_name, señalamos a Django que esas vistas están en un espacio de nombres. Ahora cuando queramos obtener una URL, usaremos el nombre del espacio de nombres y la url unidos con dos puntos.